### DISCUSSION TITLE
How to derive the bech32 stake addr from a hex stake address?

### DISCUSSION THREAD

**Discussion by bapshago** [Python-Cardano/pycardano#168] [Q&A]
I am looking for a way to derive a Bech32 stake address from a hex stake address.  Am I missing an easy way to do this with the pycardano library?

from this: "e1658e1f8524d9b7e01383c304d2075686d3e958ee1987e9e09401f84b"
nami wallet output to this  "658e1f8524d9b7e01383c304d2075686d3e958ee1987e9e09401f84b"
to this "stake1u9jcu8u9ynvm0cqns0psf5s826rd862cacvc060qjsqlsjcj6n09v"

This was my attempt. however, this outputs an address that is not a stake addr:
"addr1v9jcu8u9ynvm0cqns0psf5s826rd862cacvc060qjsqlsjcwsnfe7"

----------
`import pycardano

def convertwalletstakeaddr(hexfromnami):
	hexaddr = hexfromnami[2:]
	payment_hash = pycardano.address.VerificationKeyHash(
	    bytes.fromhex(hexaddr))
	convertedaddr = pycardano.address.Address(payment_hash).encode()
	return (convertedaddr)

test = convertwalletstakeaddr("e1658e1f8524d9b7e01383c304d2075686d3e958ee1987e9e09401f84b")
print(test) '

**Comment by cffls** [comment]
You can put the pub key hash as staking part, try this:

```
convertedaddr = pycardano.address.Address(staking_part=payment_hash).encode()
```

**Comment by bapshago** [comment]
That was it.

Thanks!
